### PR TITLE
update markdupe module + refactor mapping.nf accordingly

### PR DESCRIPTION
**Pull request by projectoriented**
## PR checklist
Updated Picard MarkDuplicates module so we no longer need samtools to index, instead utilize inherent `--CREATE_INDEX` flag from picard.
- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/raredisease/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/raredisease _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,singularity`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by projectoriented** [comment]
I forgot to do another test to ensure this works. PR paused, sorry for confusion.

File: subworkflows/nf-core/mapping.nf [hunk 1]
```diff
@@ -40,23 +39,23 @@ workflow MAPPING {
             new_meta = meta.clone()
             new_meta.id = new_meta.id.split('_')[0]
             [new_meta, bam]
-        }.groupTuple().branch{
+        }.groupTuple(by: 0).branch{
```
Comment (projectoriented) [inline subworkflows/nf-core/mapping.nf:42]: `by: 0` is the default setting but defining it explicitly for reviewers

File: subworkflows/nf-core/mapping.nf [hunk 2]
```diff
@@ -36,27 +35,30 @@ workflow MAPPING {
...
+            new_meta = meta.clone()                 // clone to avoid overriding the global meta
+            new_meta.id = new_meta.id.split('_')[0] // access the .id attribute of meta to split samplename_lane into samplename
+            [new_meta, bam]}                        // end the closure to return newly modified channel
+        .groupTuple(by: 0)                          // group them bam paths with the same [ samplename, [[bam path], [bam path], ..] ]
+        .branch{                                    // branch the channel into multiple channels (single, multiple) depending on size of list
             single: it[1].size() == 1
             multiple: it[1].size() > 1
-        }.set{ bams_to_merge }
+            }
+        .set{ bams }                                // create a new multi-channel named bams
```
Comment (projectoriented) [inline subworkflows/nf-core/mapping.nf:48]: @jemten I changed the formatting + applied comments! 🤓 🎉

Comment (jemten) [reply subworkflows/nf-core/mapping.nf]: Now it's a bit easier for me to understand 😄

**Review by jemten — APPROVED** [review]

**Pull request merged**